source/lib/rules/files-property.ts

```diff
@@ -36,3 +39,29 @@ export default (context: Context): Diagnostic[] => {
 		}
 	];
 };
+
+function processGitIgnoreStylePatterns(patterns: string[]): string[] {
+	const processedPatterns = patterns.reduce((includedPatterns, pattern) => {
+		const exclusionMarkers = pattern.match(/^!+/);
```

Comment:
I would prefer using `.filter` and `.startsWith` here.

Reply:
`.startsWith()` won't cut it here, we don't know how many markers there are...